packages/map-template/src/hooks/useMapClick.js

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+import mapTypeState from '../atoms/mapTypeState';
+import { useRecoilValue } from 'recoil';
+
+const MAP_PROVIDERS = {
+    MAPBOX: 'mapbox',
+    GOOGLE: 'google'
+}
+
+export default function useMapClick(mapsIndoorsInstance) {
+    const mapType = useRecoilValue(mapTypeState);
+    const [clickedOutside, setClickedOutside] = useState(false);
+    const map = mapsIndoorsInstance.getMap();
+
+    useEffect(() => {
+
+        if (mapType === MAP_PROVIDERS.MAPBOX) {
+            const handleMapClick = (clickResult) => {
+                const features = mapsIndoorsInstance.getMap().queryRenderedFeatures(clickResult.point);
+
+                if (features.length) {
+                    // Clicked on venue
+                    // TODO Remove console.log after review
+                    console.log('Clicked on MapsIndoors data:', features[0]);
+                    setClickedOutside(false);
+                } else if (!features.length) {
+                    setClickedOutside(true);
+                    // Clicked outside venue
+                    // TODO Remove console.log after review
+                    console.log('Clicked Nothing');
+                }
+            };
+            // Add event listener
+            map.on('click', handleMapClick);
+        }
+
+        if (mapType === MAP_PROVIDERS.GOOGLE) {
+            mapsIndoorsInstance.addListener('click', (clickResult) => {
+                setClickedOutside(false);
+                // TODO Remove console.log after review
+                console.log('Clicked on a MapsIndoors location:', clickResult);
+            });
+            map.addListener('click', (clickResult) => {
+                // TODO Remove console.log after review
+                setClickedOutside(true);
+                console.log('Clicked on a Google Maps Place:', clickResult);
+            });
+        }
+
```

Comment:
We need to remove any set event listeners in a return statement for this useEffect. If not, we risk generating multiple event listeners and/or memory leaks.